File: back-end/src/main/java/es/ull/project/application/service/operation/OperationDeleteService.java
package es.ull.project.application.service.operation;

import java.util.Optional;

import org.springframework.beans.factory.annotation.Autowired;

import es.ull.project.application.repository.OperationRepository;
import es.ull.project.application.usecase.operation.OperationDeleteUseCase;
import es.ull.project.domain.valueobject.Operation;
import es.ull.project.domain.valueobject.OperationName;

public class OperationDeleteService implements OperationDeleteUseCase {

    @Autowired
    private OperationRepository repository;

    @Override
    public Optional<Operation> deleteOperation(OperationName operationName) {
        return Optional.ofNullable(repository.delete(operationName));
    }
}

File: front-end/src/application/service/operation/operation-delete-service.ts
import { Either } from '@ull-tfg/ull-tfg-typescript';
import type { DataError, UllUUID } from '@ull-tfg/ull-tfg-typescript';
import type { OperationDeleteUseCase } from '../../usecase/operation/operation-delete-use-case';
import { Operation } from '../../../domain/value-object/operation';
import type { OperationRepository } from '../../repository/operation-repository';
import { OperationName } from '@/domain/value-object/operation-name';

export class OperationDeleteService implements OperationDeleteUseCase {

    constructor(private readonly operationRepository: OperationRepository) { }

    async deleteOperation(name: OperationName): Promise<Either<DataError, Operation>> {
        return this.operationRepository.deleteOperation(name); 
    }
}